diff -ruN ../2048/app/Game.hpp ../2048_improved/app/Game.hpp
--- ../2048/app/Game.hpp	2024-08-13 19:39:07.776606000 +0200
+++ ../2048_improved/app/Game.hpp	2024-08-15 19:59:37.087456650 +0200
@@ -15,10 +15,10 @@
 
   public:
     void init();
-    void drawHBarrier(int zeroSpaceLen);
+    static void drawHBarrier(int zeroSpaceLen);
     void drawGameBoard();
     void update();
-    void fillRandPos(GameBoard &gb);
+    static void fillRandPos(GameBoard &gb);
     void checkFinish();
 
   public:
@@ -26,7 +26,7 @@
     int getKey() const;
     void setPoint(int point);
     int getPoint() const;
-    void setGameBoard(GameBoard gb);
+    void setGameBoard(const GameBoard &gb);
     GameBoard getGameBoard() const;
     bool isFinish() const;
 
diff -ruN ../2048/app/KeyPushManager.hpp ../2048_improved/app/KeyPushManager.hpp
--- ../2048/app/KeyPushManager.hpp	2024-08-15 20:09:18.367410945 +0200
+++ ../2048_improved/app/KeyPushManager.hpp	2024-08-15 19:36:46.627607540 +0200
@@ -6,7 +6,7 @@
 class KeyPushManager
 {
   public:
-    KeyPushManager(Game &game);
+    explicit KeyPushManager(Game &game);
 
   public:
     void waitKeyPush();
Binary files ../2048/bin/2048 and ../2048_improved/bin/2048 differ
diff -ruN ../2048/src/Game.cpp ../2048_improved/src/Game.cpp
--- ../2048/src/Game.cpp	2024-08-05 18:50:49.750714000 +0200
+++ ../2048_improved/src/Game.cpp	2024-08-15 19:54:33.176637866 +0200
@@ -6,6 +6,7 @@
 
 Game::Game()
 {
+  m_key = -1;
   m_point = 0;
   m_isFinish = false;
 
@@ -203,7 +204,7 @@
   return m_gameBoard;
 }
 
-void Game::setGameBoard(Game::GameBoard gb)
+void Game::setGameBoard(const Game::GameBoard &gb)
 {
   m_gameBoard = gb;
 }
diff -ruN ../2048/src/Kbhit.cpp ../2048_improved/src/Kbhit.cpp
--- ../2048/src/Kbhit.cpp	2024-08-05 18:58:28.036697000 +0200
+++ ../2048_improved/src/Kbhit.cpp	2024-08-15 17:31:36.294222614 +0200
@@ -19,26 +19,6 @@
   tcsetattr(0, TCSANOW, &initial_settings);
 }
 
-int Kbhit::kbhit()
-{
-  unsigned char ch;
-  int nread;
-
-  if (peek_character != -1) return 1;
-  new_settings.c_cc[VMIN]=0;
-  tcsetattr(0, TCSANOW, &new_settings);
-  nread = read(0,&ch,1);
-  new_settings.c_cc[VMIN]=1;
-  tcsetattr(0, TCSANOW, &new_settings);
-
-  if (nread == 1)
-  {
-    peek_character = ch;
-    return 1;
-  }
-  return 0;
-}
-
 int Kbhit::getch()
 {
   char ch;
diff -ruN ../2048/src/KeyPushManager.cpp ../2048_improved/src/KeyPushManager.cpp
--- ../2048/src/KeyPushManager.cpp	2024-08-15 19:54:48.664044685 +0200
+++ ../2048_improved/src/KeyPushManager.cpp	2024-08-15 19:54:13.671114403 +0200
@@ -1,8 +1,9 @@
 #include "../app/KeyPushManager.hpp"
 
-KeyPushManager::KeyPushManager(Game &game) : game(game)
+KeyPushManager::KeyPushManager(Game &game) : game(game), gb(game.getGameBoard())
 {
-  gb = game.getGameBoard();
+  m_key = -1;
+  m_isGameBoardMoved = false;
 }
 
 void KeyPushManager::waitKeyPush()
